_posts/en/newsletters/2020-06-10-newsletter.md

```diff
@@ -0,0 +1,387 @@
+    This second transaction is also BIP143 invalid because one of its
+    signatures commits to an incorrect amount.  However, each of the two
+    transactions also had one valid signature and the compromised
+    software is able to use those two valid signatures to synthesize a
+    valid transaction that overpays its fees.
+
+    ![Fee overpayment attack illustration](/img/posts/2020-06-fee-overpayment-attack.dot.png)
+
+    The worst case is an attack that prompts the user to sign *n* times
+    for a wallet that controls *n* UTXOs and which results in spending
+    all of the wallet's funds to fees except for the legitimate payment
+    amount.
+
+    Similar to the other attack described in Sander's 2017 email (see
+    [Newsletter #97][news97 spk commit]), this attack only affects
+    *stateless signers*, such as hardware wallets, which depend on an
+    external system to tell them about the UTXOs they control.
+    Networked wallets which track the amounts of their received UTXOs
+    won't sign for incorrect UTXO amounts and so aren't affected by this
+    attack.  Of course, networked wallets are vulnerable to other
+    attacks, which is why hardware wallets can enhance user security.
+
+    This past week, Trezor [announced][trezor post] that Saleem Rashid
+    rediscovered this vulnerability about three months ago.  In
+    response, Trezor has updated its firmware to require copies of
+    previous transactions for segwit UTXOs the same way it requires them
+    for legacy UTXOs.  This has broken compatibility with several
+    wallets that interface with Trezor devices, either directly or
+    through [HWI][topic hwi].  In cases where wallets have full copies
+    of the previous transactions or are able to obtain them, restoring
+    compatibility should just be a matter of updating the wallet code.
+    In other cases where wallets may not store full copies of previous
+    transactions that paid the wallet, this may require redesigning the
+    wallet to store that data plus require rescanning the block chain
+    for past wallet transactions.  Software that creates or updates
+    [Partially Signed Bitcoin Transactions][topic psbt] (PSBTs) will
+    need to be [upgraded][chow full input tx] to include the full copies
+    of previous transactions into the PSBTs.  For cases where PSBTs are
+    stored in size-constrained
+    media (e.g., QR codes, see [Newsletter #96][news96 qr]), the
+    potentially significant increase in data size may require abandoning
+    the protocol or switching to higher-capacity media.
+
+    Ledger has made a similar change, although their [support
+    article][ledger update] says they only display a warning when
+    spending segwit UTXOs without full copies of their previous
+    transactions.  Although requiring previous transactions does
+    maximize user safety against this attack, there are also arguments
+    for allowing hardware wallets to optionally continue to use segwit's
```

Comment:
perhaps s/to use/using/ to align with "rather than breaking"

Reply:
Bitcoin Core PR opened yesterday on this by Andrew Chow: https://github.com/bitcoin/bitcoin/pull/19215

As well as in HWI: https://github.com/bitcoin-core/HWI/pull/340